crates/evm/src/query.rs

```diff
@@ -348,13 +349,23 @@ impl<C: sov_modules_api::Context> Evm<C> {
         block_id: Option<BlockId>,
         working_set: &mut WorkingSet<C>,
     ) -> RpcResult<reth_primitives::Bytes> {
+        let cfg = self
+            .cfg
+            .get(working_set)
+            .expect("EVM chain config should be set");
+        let (_, current_spec) = cfg.spec.last().expect("Spec should be set");
```

Comment:
this won't work actually, we don't update cfg, it was a mistake to impl. it like that. see the same file in my pr

Reply:
put todo, I'll do it in my pr